packages/vendure-plugin-accept-blue/src/api/accept-blue-service.ts

```diff
@@ -68,31 +79,186 @@ export class AcceptBlueService {
     const customer = await client.getOrCreateCustomer(
       order.customer.emailAddress
     );
+    if (!order.customer?.customFields?.activeBlueCustomerId) {
+      await this.customerService.update(ctx, {
+        id: order.customer?.id,
+        customFields: { activeBlueCustomerId: customer.id },
+      });
+    }
     // Save payment method
-    const paymentMethod = await client.getOrCreatePaymentMethod(
+    const paymentMethod = await client.getOrCreateCardPaymentMethod(
       customer.id,
       ccDetails
     );
-    return await this.payWithSavedPaymentMethod(
+    return await this.payWithCardPaymentMethodDetails(
+      ctx,
+      order,
+      amount,
+      client,
+      paymentMethod.id,
+      ccDetails
+    );
+  }
+
+  async payWithToken(
+    ctx: RequestContext,
+    order: Order,
+    amount: number,
+    client: AcceptBlueClient,
+    ccDetails: TokenPaymentMethodInput
+  ): Promise<HandlePaymentResult> {
+    if (!order.customer) {
+      throw new UserInputError(`Order must have a customer`);
+    }
+    const customer = await client.getOrCreateCustomer(
+      order.customer.emailAddress
+    );
+    if (!order.customer?.customFields?.activeBlueCustomerId) {
+      await this.customerService.update(ctx, {
+        id: order.customer?.id,
+        customFields: { activeBlueCustomerId: customer.id },
+      });
+    }
+    // Save payment method
+    const methods = await client.getPaymentMethods(
+      order.customer.customFields.activeBlueCustomerId
+    );
+    const paymentMethod = methods.find((method) =>
+      isSameCard(ccDetails, method as AcceptBlueCardPaymentMethod)
+    );
+    if (!paymentMethod) {
+      throw new UserInputError(
+        'No Saved Credit Card with this toknized nonce code'
+      );
+    }
+    return await this.payWithCardPaymentMethodDetails(
       ctx,
       order,
       amount,
       client,
-      paymentMethod.id
+      paymentMethod.id,
+      ccDetails
     );
   }
 
+  async payWithCheck(
+    ctx: RequestContext,
+    order: Order,
+    amount: number,
+    client: AcceptBlueClient,
+    ccDetails: CheckPaymentInput
```

Comment:
`cc` stands for creditcard, so in this case we can rename it to `checkDetails` or something?

Same goes for `payWithToken` input above